kernel/init.c

```diff
@@ -249,6 +249,7 @@ static void bg_thread_main(void *unused1, void *unused2, void *unused3)
 void __weak main(void)
 {
 	/* NOP default main() if the application does not provide one. */
+	__asm__ volatile("nop");
```

Comment:
the problem removing that is that this will break all applications that are using main() as a start point (Am I missing something ?).
I've checked x86/arc/arm, CI will probably catch if this is a problem in the other archs.

Reply:
Yeah, seems like defining an _arch_nop() primitive would be a better idea, though significantly more typing.

At some point you'll need to sit me down with the exact text of MISRA-C rule 2.2 to convince me this is really the intent.  An empty function isn't an error, it's not undefined, it's not unexercised or untested.  It's a live function that has explicit behavior (nothing) that gets executed at runtime (though perhaps optimized away) and can be tested.  Where we've discussed this before it seems like the term "dead code" gets thrown around for these, and that's wrong -- they're not dead at all.

Reply:
Sure we can do that. For Misra, code that does nothing is code that should be removed - > dead code.

Reply:
What is the verdict for this change ? Should I add an _arch_nop() or this is fine ?